### ISSUE TITLE
MemcacheTest.testBulkSetGet_withManyKeys

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast#9637]
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/6683/testReport/junit/com.hazelcast.internal.ascii/MemcacheTest/testBulkSetGet_withManyKeys/

Regression

Timeout waiting for bulk values: waited 216,000 ms. Node status: Connection Status { /127.0.0.1:5703 active: true, authed: true, last read: 215,997 ms ago }
Stacktrace

net.spy.memcached.OperationTimeoutException: Timeout waiting for bulk values: waited 216,000 ms. Node status: Connection Status { /127.0.0.1:5703 active: true, authed: true, last read: 215,997 ms ago }
	at net.spy.memcached.internal.BulkGetFuture.get(BulkGetFuture.java:126)
	at net.spy.memcached.internal.BulkGetFuture.get(BulkGetFuture.java:51)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1465)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1509)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1523)

**Comment by jerrinot** [comment]
I know the root cause: with the bulk get we generate a bunch of `PartialGetCommend` at https://github.com/jerrinot/hazelcast/blob/eef987b9920f3f7b4c506e433b0d067cab9ba93c/hazelcast/src/main/java/com/hazelcast/internal/ascii/memcache/GetCommandParser.java#L34-L34

and then we publish the `EndCommand`

These commands are executed here: https://github.com/jerrinot/hazelcast/blob/f777fcc86c5a8a8a84c47d2261d8aa321f143504/hazelcast/src/main/java/com/hazelcast/internal/ascii/TextCommandServiceImpl.java#L222

`hz:text` uses multiple threads: it can pick-up the `EndCommand` before all `PartialGetCommand`s are processed -> the test hangs.

When I use just a single thread in the `hz:text` executor then the issue is gone. However I am not sure if that's the right thing to do: it can lead to:
1. bad performance regression: think of bulk command where keys are located on different members
2. glitches in the man. centre? a long running command will block the REST interface used my the ManCen

**Comment by hasancelik** [comment]
```
net.spy.memcached.OperationTimeoutException: Timeout waiting for bulk values: waited 216,000 ms. Node status: Connection Status { /127.0.0.1:5715 active: true, authed: true, last read: 215,986 ms ago }
	at net.spy.memcached.internal.BulkGetFuture.get(BulkGetFuture.java:126)
	at net.spy.memcached.internal.BulkGetFuture.get(BulkGetFuture.java:51)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1465)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1509)
	at net.spy.memcached.MemcachedClient.getBulk(MemcachedClient.java:1523)
	at com.hazelcast.internal.ascii.MemcacheTest.testBulkSetGet_withManyKeys(MemcacheTest.java:353)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-IbmJDK1.8/165/com.hazelcast$hazelcast/testReport/com.hazelcast.internal.ascii/MemcacheTest/testBulkSetGet_withManyKeys/

**Comment by jerrinot** [comment]
fixed by #9642
